src/main/java/org/broadinstitute/hellbender/tools/evoquer/GnarlyGenotyperEngine.java

```diff
@@ -0,0 +1,492 @@
+            }
+        }
+
+        for ( final Genotype g : vc.getGenotypes() ) {
+            final String name = g.getSampleName();
+            if(g.getPloidy() != ASSUMED_PLOIDY && !isGDBnoCall(g)) {
+                throw new UserException.BadInput("This tool assumes diploid genotypes, but sample " + name + " has ploidy "
+                        + g.getPloidy() + " at position " + vc.getContig() + ":" + vc.getStart() + ".");
+            }
+            final Genotype calledGT;
+            final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(g);
+            genotypeBuilder.name(name);
+            if (isGDBnoCall(g) || g.getAllele(0).equals(Allele.NON_REF_ALLELE) || g.getAllele(1).equals(Allele.NON_REF_ALLELE)) {
+                genotypeBuilder.alleles(GATKVariantContextUtils.noCallAlleles(ASSUMED_PLOIDY));
+            }
+            else if (nonRefReturned) {
+                if (g.hasAD()) {
+                    final int[] AD = trimADs(g, targetAlleles.size());
+                    genotypeBuilder.AD(AD);
+                }
+                else if (g.countAllele(Allele.NON_REF_ALLELE) > 0) {
+                    genotypeBuilder.alleles(GATKVariantContextUtils.noCallAlleles(ASSUMED_PLOIDY)).noGQ();
+                }
+            }
+            if (g.hasPL()) {
+                if (summarizePLs) {
+                    summarizePLs(genotypeBuilder, g, vc);
+                } else {
+                    final int[] PLs = trimPLs(g, newPLsize);
+                    genotypeBuilder.PL(PLs);
+                    genotypeBuilder.GQ(MathUtils.secondSmallestMinusSmallest(PLs, 0));
+                    //If GenomicsDB returns no-call genotypes like CombineGVCFs, then we need to actually find the GT from PLs
+                    makeGenotypeCall(genotypeBuilder, GenotypeLikelihoods.fromPLs(PLs).getAsVector(), targetAlleles);
+                }
+            }
+            final Map<String, Object> attrs = new HashMap<>(g.getExtendedAttributes());
+            attrs.remove(GATKVCFConstants.MIN_DP_FORMAT_KEY);
+            //attrs.remove(GATKVCFConstants.STRAND_BIAS_BY_SAMPLE_KEY);
+            calledGT = genotypeBuilder.attributes(attrs).make();
+            mergedGenotypes.add(calledGT);
+
+            if (g.hasAnyAttribute(GATKVCFConstants.STRAND_BIAS_BY_SAMPLE_KEY)) {
+                try {
+                    @SuppressWarnings("unchecked")
+                    final List<Integer> sbbsList = (ArrayList<Integer>) g.getAnyAttribute(GATKVCFConstants.STRAND_BIAS_BY_SAMPLE_KEY);
+                    MathUtils.addToArrayInPlace(SBsum, Ints.toArray(sbbsList));
+                }
+                catch (ClassCastException e) {
+                    throw new IllegalStateException("The GnarlyGenotyper tool assumes that input variants come from " +
+                            "GenomicsDB and have SB FORMAT fields that have already been parsed into ArrayLists.");
```

Comment:
"The GnarlyGenotyper tool assumes that input variants come from GenomicsDB" -> not strictly true anymore :)